# Statistical Analysis of Fremont Bridge Crossings, Mahir Piyarali

## Introduction

This project will find the answer to the question "What is the best day to cross the Fremont Bridge?" For the purpose of this project as well as consistency, I will be defining the "best" day to cross as the day where traffic is lowest. I choose this because when traffic is lower, in theory it should be easier to cross as there are less people in the way, and therefore, is the best day to cross.

Fremont Bridge crossings data comes from the Seattle Government.

### Analysis 1

###### Looking at the total amount of crossings per day in the given time frame, to see which had the least amount of total traffic, therefore is the best day to cross

#### Dataset Preparation

I chose to prepare the data by importing it, and then creating two separate lists. One list for eastbound traffic, and one for west. The dataset is strong as it covers almost 4 years worth of daily crossings. This means that there is lots of data, and that biases should be low.

In [56]:
#the following code imports the dataset into the code
bridge_file = open("FreBridgeLab/FreBridge2.csv") 
bridge_data = []
for ii in bridge_file:
    bridge_data.append(ii);
bridge_file.close()
del(bridge_data[0]);
#the following code appends the data into seperate lists for westbound and eastbound traffic
west = []; 
east = [];
for jj in range(len(bridge_data)):
    bridge_data[jj] = bridge_data[jj].split(",");
    west.append(int(bridge_data[jj][1]));
    east.append(int(bridge_data[jj][2]));

#### Data Modeling

Simply uses addition/total sums to model the data

In [57]:
#the following code appends the westbound traffic into lists based on the day of the week
#the dataset provided has the first date (10/3/2012) as a wednesday. this is taken into consideration when creating the lists for each day of the week
monW = west[5::7] #the list "monW" is a list of daily traffic for each monday westbound
tueW = west[6::7] #the list "tueW" " " " " " " " " tuesday "
wedW = west[0::7] #and so on...
thuW = west[1::7]
friW = west[2::7]
satW = west[3::7]
sunW = west[4::7]
#the following code adds together the sum of the traffic across all of the days into a single sum
monWsum = sum(monW) #represents all of the traffic that crossed the bridge westbound on monday
tueWsum = sum(tueW) # " " " " " " " " " " " tuesday
wedWsum = sum(wedW) #and so on...
thuWsum = sum(thuW)
friWsum = sum(friW)
satWsum = sum(satW)
sunWsum = sum(sunW)
#the following code does the same to the eastbound data
monE = east[0::7]
tueE = east[1::7]
wedE = east[2::7]
thuE = east[3::7]
friE = east[4::7]
satE = east[5::7]
sunE = east[6::7]
monEsum = sum(monE)
tueEsum = sum(tueE)
wedEsum = sum(wedE)
thuEsum = sum(thuE)
friEsum = sum(friE)
satEsum = sum(satE)
sunEsum = sum(sunE)
#the following code creates TOTAL sums (both east and west) for each day of the week
totalMon = (monWsum + monEsum) #represents ALL of the traffic (both west and east) that crossed the bridge on monday within the time frame of data collection
totalTue = (tueWsum + tueEsum)
totalWed = (wedWsum + wedEsum)
totalThu = (thuWsum + thuEsum)
totalFri = (friWsum + friEsum)
totalSat = (satWsum + satEsum)
totalSun = (sunWsum + sunEsum)
#the following code then creates a list called "totalWk" within which each variable is the total amount of traffic to cross the bridge on a given day of the week within the timeframe of the data
totalWk = [totalMon,totalTue,totalWed,totalThu,totalFri,totalSat,totalSun]
print("The total amount of traffic to cross the Fremont Bridge (both east and west) is", totalWk)
print("By the metric of simply looking at all of the traffic to cross the bridge in the timeframe, it appears as if Friday is the best day to cross, because in the timeframe it had the least amount of crossings, with", min(totalWk), "crossings")

The total amount of traffic to cross the Fremont Bridge (both east and west) is [680343, 685681, 656557, 503223, 454676, 491852, 506607]
By the metric of simply looking at all of the traffic to cross the bridge in the timeframe, it appears as if Friday is the best day to cross, because in the timeframe it had the least amount of crossings, with 454676 crossings


### Analysis 2

###### Comparing the average amount of traffic per day of the week , to see which day (in a given direction) had the least amount of traffic and is therefore the best day to cross (in that given direction)

#### Dataset Preparation

I chose to prepare the data by importing it, and then creating two separate lists. One list for eastbound traffic, and one for west. The dataset is strong as it covers almost 4 years worth of daily crossings. This means that there is lots of data, and that biases should be low.

In [58]:
#the following code imports the dataset into the code, is copy and pasted from above
bridge_file = open("FreBridgeLab/FreBridge2.csv")
bridge_data = []
for ii in bridge_file:
    bridge_data.append(ii);
bridge_file.close()
del(bridge_data[0]);
#the following code appends the data into separate lists for eastbound and westbound 
west = [];
east = [];
for jj in range(len(bridge_data)):
    bridge_data[jj] = bridge_data[jj].split(",");
    west.append(int(bridge_data[jj][1]));
    east.append(int(bridge_data[jj][2]));

#### Data Modeling

Dataset is modeled by using averages, total sums, and lengths to determine the best day to cross

In [59]:
#the following code appends the westbound traffic into lists based on the day of the week
#an important note is that the dataset's first date (10/3/2012) was a wednesday. this is taken into consideration when appending data into separate lists.
mW = west[5::7] #the list "monW" is a list of daily traffic for each monday westbound
tuW = west[6::7] #the list "tueW" " " " " " " " " tuesday "
wW = west[0::7] #and so on...
thW = west[1::7]
fW = west[2::7]
saW = west[3::7]
suW = west[4::7]
#the following code creates new variables to hold the amount of each day of the week there are in the dataset
#example: there are 8 mondays in the dataset (this could be wrong, just an example!!!)
#these lengths will be used to calculate the averages
lmW = len(mW)
ltuW = len(tuW)
lwW = len(wW)
lthW = len(thW)
lfW = len(fW)
lsaW = len(saW)
lsuW = len(suW)
#the following code then creates new varables to hold the TOTAl amount of crossings for each day of the week
#example: in total, there were 18273 westbound crossings on mondays (this could be wrong, just an example!!!)
#these sums will be used to calculate the averages
mWsum = sum(mW)
tuWsum = sum(tuW) 
wWsum = sum(wW)
thWsum = sum(thW)
fWsum = sum(fW)
saWsum = sum(saW)
suWsum = sum(suW) 
#the following code then creates new variables to hold the AVERAGE amount of crossings for any given day of the week
#example: on average, there are 1928 westbound crossings on Monday (this could be wrong, just an example!!!)
avmW = round((mWsum)/(lmW)) #the "round" part of this code makes it so that the code rounds off the average to a whole number
avtuW = round((tuWsum)/(ltuW))
avwW = round((wWsum)/(lwW))
avthW = round((thWsum)/lthW)
avfW = round((fWsum)/lfW)
avsaW = round((saWsum)/lsaW)
avsuW = round((suWsum)/lsuW)
#the following code then creates a list of all of the averages for the westbound traffic by day of the week
westAvgs = [avmW,avtuW,avwW,avthW,avfW,avsaW,avsuW]
print("The average amount of traffic to cross the bridge westbound by day of the week is", westAvgs)
print("By looking at the average amount of westbound traffic per day of the week, it looks like Sunday is the best day to cross when headed westbound, as it had the lowest amount of average westbound crossings with a total of", min(westAvgs))
#the following code does the same to the eastbound data
mE = east[5::7]
tuE = east[6::7]
wE = east[0::7]
thE = east[1::7]
fE = east[2::7]
saE = east[3::7]
suE = east[4::7]
lmE = len(mE)
ltuE = len(tuE)
lwE = len(wE)
lthE = len(thE)
lfE = len(fE)
lsaE = len(saE)
lsuE = len(suE)
mEsum = sum(mE)
tuEsum = sum(tuE) 
wEsum = sum(wE)
thEsum = sum(thE)
fEsum = sum(fE)
saEsum = sum(saE)
suEsum = sum(suE) 
#the following code then creates new variables to hold the AVERAGE amount of crossings for any given day of the week
#example: on average, there are 1928 westbound crossings on Monday (this could be wrong, just an example!!!)
avmE = round((mEsum)/(lmE)) #the "round" part of this code makes it so that the code rounds off the average to a whole number
avtuE = round((tuEsum)/(ltuE))
avwE = round((wEsum)/(lwE))
avthE = round((thEsum)/lthE)
avfE = round((fEsum)/lfE)
avsaE = round((saEsum)/lsaE)
avsuE = round((suEsum)/lsuE)
#the following code then creates a list of all of the averages for the westbound traffic by day of the week
eastAvgs = [avmE,avtuE,avwE,avthE,avfE,avsaE,avsuE]
print("The average amount of traffic to cross the bridge westbound by day of the week is", eastAvgs)
print("By looking at the average amount of westbound traffic per day of the week, it looks like Sunday is the best day to cross when headed westbound, as it had the lowest amount of average westbound crossings with a total of", min(eastAvgs))


The average amount of traffic to cross the bridge westbound by day of the week is [1698, 1831, 1834, 1749, 1557, 878, 849]
By looking at the average amount of westbound traffic per day of the week, it looks like Sunday is the best day to cross when headed westbound, as it had the lowest amount of average westbound crossings with a total of 849
The average amount of traffic to cross the bridge westbound by day of the week is [1533, 1634, 1629, 1531, 1375, 717, 672]
By looking at the average amount of westbound traffic per day of the week, it looks like Sunday is the best day to cross when headed westbound, as it had the lowest amount of average westbound crossings with a total of 672


## Final Results / Data Analysis

###### Breaking down Analysis 1
Analysis 1 is simple, it simply adds all of the traffic on each day of the week together, and then uses that total amount of traffic to determine which day is the best day to cross. However, this is nowhere near specific enough to be useful. That's because though Friday may have the lowest number of total crossings (45476), it doesn't help determine if westbound or eastbound is better on a given day. It also doesn't take into account that perhaps everyone only goes eastbound on Friday. Therefore yes, going westbound would be best, and yes, the total amount of people crossing is low, but it fails is properly determining the best day to cross.


###### Breaking down Analysis 2

Analysis 2 is much much much more thorough in determining the best day to cross. It first looks at all of the westbound traffic, and breaks it up by each day. It then calculates the average amount of westbound traffic for a given day (average crossings on mondays headed west was 1698). The analysis then uses these averages to determine which day had the lowest amount of average crossings, and that was the metric used to determine the "best" day to cross the bridge headed westbound, which turned out to be Sunday, with a total of 849 westbound crossings on average. Coincidentally, Sunday was also the best day to cross headed east, with a total of 672 crossings on average. This actually makes some sense, as there are NFL games on Sundays, Church meets on Sundays, and generally speaking, Sunday is the laziest day of the week

###### Remaining questions, things unanswered, additional conclusions

Unfortunately, the dataset given doesn't account for days in which traffic could be exceptionally low. And when looking in the data, there for sure are some outliers. This could mean that by pure coincidence in a calendar year, all of the major holidays (Christmas, Thanksgiving, etc.) fell on a single day of the week, therefore biasing some of the results. 

What I also find interesting is that though Analysis 1 showed that Friday had the lowest amount of total overall crossings, it turns out that the daily averages show that for either westbound or eastbounf traffic, the best day of the week to cross is actually Sunday. I wonder why it turned out this way... perhaps the dataset had more fridays than it did sundays?

### Best day to cross westbound: Sunday, Best day to cross eastbound: Sunday